Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -51,17 +52,36 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def pickable(cls, commits):
-        filtered_commits = []
+    def pickable(cls, commits, repository, commits_bundle=None):
```

Comment:
What is a `commits_bundle`?

Reply:
I'm open to other names for this object.

Basically, it is often the case that another commit has the information that defines a relationship for the commit in question. Take, for example, a commit which is reverted by a subsequent commit. The subsequent commit defines the relationship. The `commits_bundle` object basically has a bunch of pre-computed commit relationships in it so we can just look them up by key instead of searching through all commits. It basically turns `pickable` from `O(n^2)` to `O(n)`.

Reply:
I think I'd suggest a name like `CommitsStory` rather than `CommitsBundle`, "bundle" didn't imply that the contained commits have a meaningful relationship.